# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: caps/OriginAttributes.cpp
# Commit: 7c4b003d1e48
# Full Hash: 7c4b003d1e48ac2e847191fe36ce4c39b9280d54
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2024-12-18 04:27:40
# Regressor Bug: 1915355
# File Overlap Count: 1
# Description:
#   Bug 1915355 - Part 4: Implement the third-party cookie blocking excpetion in CookieService. r=cookie-reviewers,baku,valentin
#   
#   The patch integrates the third-party cookie blocking excpetion in
#   CookieService.
#   
# ==============================================================================

diff -r 36163490219b -r 7c4b003d1e48 caps/OriginAttributes.cpp
--- a/caps/OriginAttributes.cpp	Tue Dec 17 21:44:37 2024 +0000
+++ b/caps/OriginAttributes.cpp	Tue Dec 17 21:44:37 2024 +0000
@@ -61,7 +61,7 @@
                                         bool aForeignByAncestorContext,
                                         bool aIsFirstPartyEnabled, bool aForced,
                                         bool aUseSite,
-                                        nsString OriginAttributes::*aTarget,
+                                        nsString OriginAttributes::* aTarget,
                                         OriginAttributes& aOriginAttributes) {
   nsresult rv;
 
@@ -514,4 +514,23 @@
   return fieldIndex > 1;
 }
 
+/* static */
+bool OriginAttributes::ExtractSiteFromPartitionKey(
+    const nsAString& aPartitionKey, nsAString& aOutSite) {
+  nsAutoString scheme, host;
+  int32_t port;
+  bool unused;
+  if (!ParsePartitionKey(aPartitionKey, scheme, host, port, unused)) {
+    return false;
+  }
+
+  if (port == -1) {
+    aOutSite.Assign(scheme + u"://"_ns + host);
+  } else {
+    aOutSite.Assign(scheme + u"://"_ns + host + u":"_ns);
+    aOutSite.AppendInt(port);
+  }
+  return true;
+}
+
 }  // namespace mozilla